### ISSUE TITLE
为什么现在的这个albert-chinese-ner没有以前的bert-chinese-ner准确率高呢？你对比过吗？

### ISSUE THREAD

**Issue by sxk000** [ProHiryu/albert-chinese-ner#25]
你好，我用同样的参数跑了一下albert-chinese-ner和上一个bert-chinese-ner，但是跑出来的结果后者好于前者，论文上不是说albert比bert好一些吗？为什么不是这样呢？下面是我跑出来的结果：

albert-chinese-ner跑的命令和结果如下：
用的预训练向量地址：https://storage.googleapis.com/albert_zh/albert_base_zh.zip
命令：
python3 albert_ner.py --task_name ner --do_train true --do_eval true --data_dir data --vocab_file albert_config/vocab.txt --bert_config_file ../../albert_base_zh/albert_config_base.json --max_seq_length 128 --train_batch_size 64 --learning_rate 2e-5 --num_train_epochs 3 --output_dir albert_base_ner_checkpoints
结果：
INFO:tensorflow:***** Eval results albert_base_ner_checkpoints/model.ckpt-2374 *****
INFO:tensorflow:  eval_f = 0.9456498
INFO:tensorflow:  eval_precision = 0.95313764
INFO:tensorflow:  eval_recall = 0.93832487
INFO:tensorflow:  global_step = 2374
INFO:tensorflow:  loss = 11.592924

bert-chinese-ner跑的命令和结果如下：
用的预训练向量地址：https://storage.googleapis.com/bert_models/2018_11_03/chinese_L-12_H-768_A-12.zip
命令：
python3 BERT_NER.py --task_name ner --do_train true --do_eval true --data_dir data --vocab_file vocab.txt --bert_config_file checkpoint/bert_config.json --init_checkpoint checkpoint/bert_model.ckpt --max_seq_length 128 --train_batch_size 64 --learning_rate 2e-5 --num_train_epochs 3 --output_dir output/result_dir/
结果：
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 2374: output/result_dir/model.ckpt-2374
INFO:tensorflow:evaluation_loop marked as finished
INFO:tensorflow:***** Eval results *****
INFO:tensorflow:  eval_f = 0.96657985
INFO:tensorflow:  eval_precision = 0.9670561
INFO:tensorflow:  eval_recall = 0.96620923
INFO:tensorflow:  global_step = 2374
INFO:tensorflow:  loss = 8.526184

训练时间bert也比albert快，训练时每一个examples/sec耗时如下
albert耗时平均:     INFO:tensorflow:examples/sec: 66.2794
bert耗时平均:        INFO:tensorflow:examples/sec: 58.3886

为什么会出现这样的结果呢？

**Comment by JamesGu14** [comment]
wondering which GPU did you use to perform the train? I'm with 2080Ti, seems my test has a smaller batch_size but faster processing speed

my test on MSRA corpus, ALBert base is 150 samples/s and outcome eval_f is 0.9475

**Comment by sxk000** [comment]
> wondering which GPU did you use to perform the train? I'm with 2080Ti, seems my test has a smaller batch_size but faster processing speed
> 
> my test on MSRA corpus, ALBert base is 150 samples/s and outcome eval_f is 0.9475

i use p40

**Comment by ProHiryu** [comment]
尝试了几次呢？ 是否每次都出现这种结果？

**Comment by sxk000** [comment]
> 尝试了几次呢？ 是否每次都出现这种结果？

跑了三次，结果都差不多

**Comment by ProHiryu** [comment]
这个可能跟不同的模型参数和训练参数有关，虽然我看你写出来的都是一样，但是不代表每个参数都一致，还有bert基础模型的更新时间可能也会影响，不过如果相差不大也没什么关系

**Comment by sxk000** [comment]
> 这个可能跟不同的模型参数和训练参数有关，虽然我看你写出来的都是一样，但是不代表每个参数都一致，还有bert基础模型的更新时间可能也会影响，不过如果相差不大也没什么关系

使用bert的参数如下：
{
  "attention_probs_dropout_prob": 0.1, 
  "directionality": "bidi", 
  "hidden_act": "gelu", 
  "hidden_dropout_prob": 0.1, 
  "hidden_size": 768, 
  "initializer_range": 0.02, 
  "intermediate_size": 3072, 
  "max_position_embeddings": 512, 
  "num_attention_heads": 12, 
  "num_hidden_layers": 12, 
  "pooler_fc_size": 768, 
  "pooler_num_attention_heads": 12, 
  "pooler_num_fc_layers": 3, 
  "pooler_size_per_head": 128, 
  "pooler_type": "first_token_transform", 
  "type_vocab_size": 2, 
  "vocab_size": 21128
}

使用albert的参数如下：
{
  "attention_probs_dropout_prob": 0.0,
  "directionality": "bidi", 
  "hidden_act": "gelu", 
  "hidden_dropout_prob": 0.0,
  "hidden_size": 768,
  "embedding_size": 128,
  "initializer_range": 0.02, 
  "intermediate_size": 3072 ,
  "max_position_embeddings": 512, 
  "num_attention_heads": 12,
  "num_hidden_layers": 12,
  "pooler_fc_size": 768,
  "pooler_num_attention_heads": 12,
  "pooler_num_fc_layers": 3, 
  "pooler_size_per_head": 128, 
  "pooler_type": "first_token_transform", 
  "type_vocab_size": 2, 
  "vocab_size": 21128,
   "ln_type":"postln"
}

参数基本都是一样的

**Comment by ProHiryu** [comment]
不知道bert是否更新了模型，可能基于新的语料有新的预训练模型

**Comment by sxk000** [comment]
> 不知道bert是否更新了模型，可能基于新的语料有新的预训练模型

嗯，好的，谢谢你！
